skimage/util/dtype.py

```diff
@@ -465,10 +467,36 @@ def img_as_float(image, force_copy=False):
     and can be outside the ranges [0.0, 1.0] or [-1.0, 1.0].
 
     """
-    return _convert(image, np.floating, force_copy)
+    return _img_as_float(image, force_copy, preserve_range=False)
+
+
+def _img_as_float(image, force_copy=False, *, preserve_range=True, dtype=None):
```

Comment:
I think `_rescale_to_float` would be the clearer and more consistent name.

Reply:
_img_as_float has been removed and _convert is just used directy now.